docs/cpp.md

```diff
@@ -3,8 +3,86 @@
 Users are typically migrating from [cc_image](https://github.com/bazelbuild/rules_docker#cc_image)
 in rules_docker.
 
-TODO: document how to build an image
+## An example of packaging a simple C++ program
 
-## Example
+Using a minimal example C++ program `example.cc`:
+```cpp
+#include <iostream>
 
-An example is needed! Please consider contributing one. :pray:
+int main(){
+    std::cout<<"This is a C++ example!"<<std::endl;
+}
+```
+
+To make image and container for this program, the `BUILD.bazel` file would look like this:
+```python
+load("@rules_oci//oci:defs.bzl", "oci_image", "oci_tarball")
+load("@rules_cc//cc:defs.bzl", "cc_binary")
+load("@rules_pkg//pkg:tar.bzl", "pkg_tar")
+
+package(default_visibility = ["//visibility:public"])
+
+# Normal cc_binary
+cc_binary(
+    name  = "example_binary",
+    srcs = [
+        "example.cc",
+    ]
+)
+
+# Packaging the binary into tar, which is needed by oci_image rule
+pkg_tar(
+    name = "tar",
+    srcs = [":example_binary"],
+)
+
+# Making image
+# C++ programs usually need some fundamental libraries such as glibc, libstdc++, etc.
+# `distroless_base` image does not provide them and C++ programs will complain about
```

Comment:
maybe distroless_cc does?

Reply:
Makes sense. Changed the example to use `gcr.io/distroless/cc-debian12`